src/main/scala/org/zalando/kanadi/api/Subscriptions.scala

```diff
@@ -540,14 +547,7 @@ case class Subscriptions(baseUri: URI, oAuth2TokenProvider: Option[OAuth2TokenPr
         consumerGroupCheck && idCheck
       }
 
-      head = collect.headOption
-
-      createIfEmpty <- {
-        head match {
-          case Some(subscription) => Future.successful(subscription)
-          case None               => create(subscription)
-        }
-      }
+      createIfEmpty <- collect.headOption.map(Future(_)).getOrElse(create(subscription))
```

Comment:
Just a minor note, it's better to use the Future.successful constructor here, since we're not expecting some async computation to still be done.

Reply:
Agreed, have to do it in a separate PR though, this is just cherry picked commits